primedev/mods/modweaponvars.cpp

```diff
@@ -0,0 +1,255 @@
+
+#include "client/weaponx.h"
+#include "server/weaponx.h"
+#include "modweaponvars.h"
+#include "squirrel/squirrel.h"
+#include "client/r2client.h"
+#include "server/r2server.h"
+
+typedef bool (*calculateWeaponValuesType)(int bitfield, void* weapon, void* weaponVarLocation);
+typedef char* (*get2ndParamForRecalcModFuncType)(void* weapon);
+
+template <ScriptContext context> get2ndParamForRecalcModFuncType get2ndParamForRecalcModFunc;
+template <ScriptContext context> calculateWeaponValuesType _CalculateWeaponValues;
+template <ScriptContext context> WeaponVarInfo* weaponVarArray;
+template <ScriptContext context> std::vector<SQObject> weaponModCallbacks;
+
+template <ScriptContext context> bool IsWeapon(void** ent)
+{
+	if (context == ScriptContext::SERVER)
+		return *ent == CWeaponX_vftable;
+	else
+		return *ent == C_WeaponX_vftable;
+}
+
+AUTOHOOK_INIT()
+
+// RENAME THIS BEFORE MERGE (REVIEWERS IF YOU STILL SEE THIS BLOCK IT NOWNOWNOWNOWNOW)
+AUTOHOOK(Cl_WeaponTick, client.dll + 0x59D1E0, bool, __fastcall, (C_WeaponX * weapon))
```

Comment:
yeah, since we got rid of the other hook (we no longer need it) i can just call this Cl_WeaponTick...

the name might be wrong tho? ill be adding a comment for now